SortingAlgorithms/Algorithms/Searching/BinarySearch.cs

```diff
@@ -11,44 +11,44 @@ namespace SortingAlgorithms.Algorithms.Searching
 {
     public class BinarySearch : ISearchingAlgorithm
     {
-        public void Search(int[] array, int number)
+        public void Search<T>(T[] array, T item) where T : IComparable
         {
             if (SortedChecker.IsSorted(array) is false)
             {
                 ISortingAlgorithm sortingAlgorithm = new MergeSort();
```

Comment:
Better to use it as property and set from constructor
The same for SortedChecker, make it dynamic and set from ctor